src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+`;
+
+const PanelWarning = styled(Text).attrs(({ theme: { colors } }) => ({
+  color: colors.yellowFavorite,
+  lineHeight: 'normal',
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(-20, 0, 20, 0)};
+`;
+
+const PanelError = styled(Text).attrs(({ theme: { colors } }) => ({
+  color: colors.red,
+  lineHeight: 'normal',
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(-20, 0, 20, 0)}
+`;
+
+const GasTrendHeader = styled(Text).attrs(({ theme: { colors }, color }) => ({
+  color: color || colors.appleBlue,
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(0, 5, 0, 0)};
+`;
+
+const PanelColumn = styled(Column).attrs(() => ({
+  justify: 'center',
+}))``;
+
+const Label = styled(Text).attrs(({ size }) => ({
+  size: size || 'lmedium',
+}))`
+  ${({ weight }) => fontWithWidth(weight || fonts.weight.semibold)}
+`;
+
+const GAS_FEE_INCREMENT = 1;
+const MAX_BASE_FEE_RANGE = [1, 3];
+const MINER_TIP_RANGE = [1, 2];
+
+export default function FeesPanel({
+  currentGasTrend,
+  colorForAsset,
+  onCustomGasFocus,
+}) {
+  const {
+    selectedGasFee,
+    currentBlockParams,
+    gasFeeParamsBySpeed,
+    updateToCustomGasFee,
+    updateGasFeeOption,
+    selectedGasFeeOption,
+  } = useGas();
+  const { navigate } = useNavigation();
+  const { isDarkMode } = useTheme();
+
+  const [customMaxPriorityFee, setCustomMaxPriorityFee] = useState(
+    get(selectedGasFee, 'gasFeeParams.maxPriorityFeePerGas.gwei', 0)
+  );
+  const [customMaxBaseFee, setCustomMaxBaseFee] = useState(
+    get(selectedGasFee, 'gasFeeParams.maxFeePerGas.gwei', 0)
+  );
+  const [maxPriorityFeeWarning, setMaxPriorityFeeWarning] = useState(null);
+  const [maxPriorityFeeError, setMaxPriorityFeeError] = useState(null);
+
+  const [maxBaseFeeWarning, setMaxBaseFeeWarning] = useState(null);
+  const [maxBaseFeeError, setMaxBaseFeeError] = useState(null);
+  const [feesGweiInputFocused, setFeesGweiInputFocused] = useState(false);
+
+  const maxBaseWarningsStyle = useAnimatedStyle(() => {
+    const display = !!maxBaseFeeError || !!maxBaseFeeWarning;
+    return {
+      transform: [{ scale: display ? 1 : 0 }],
+    };
+  });
+
+  const maxPriorityWarningsStyle = useAnimatedStyle(() => {
+    const display = !!maxPriorityFeeError || !!maxPriorityFeeWarning;
+    return {
+      transform: [{ scale: display ? 1 : 0 }],
+    };
+  });
+
+  const selectedOptionIsCustom = useMemo(
+    () => selectedGasFee?.option === gasUtils.CUSTOM,
+    [selectedGasFee?.option]
+  );
+
+  const { maxFee, currentBaseFee, maxBaseFee, maxPriorityFee } = useMemo(() => {
+    const maxFee = get(selectedGasFee, 'gasFee.maxFee.native.value.display', 0);
+
+    const currentBaseFee = get(currentBlockParams, 'baseFeePerGas.gwei', 0);
+    let maxBaseFee;
+    if (selectedOptionIsCustom) {
+      // block more than 2 decimals on gwei value
+      const decimals = Number(customMaxBaseFee) % 1;
+      maxBaseFee =
+        `${decimals}`.length > 4
+          ? parseInt(customMaxBaseFee, 10)
```

Comment:
this seems arbitrary to do a parseInt (what if the customMaxBaseFee in gwei was 4.9556 etc, the parseInt for something with decimals of length > 4 would just truncate to gwei as 4)

Reply:
I did this because we're only showing int numbers for the max base fee, but if the user wants to put a decimal it will allow it... thinking it twice we may want to just go for ints tho. WDYT?

Reply:
we also don't want to have fixed decimals when we're displaying an int

Reply:
conclusion: using `toFixedDecimals(customMaxBaseFee, 0)`
confirm with Christian that having only integers is cool for the base fee